src/apps/activity_assignments/service.py

```diff
@@ -64,6 +73,79 @@ async def create_many(
             for assignment in assignment_schemas
         ]
 
+    async def check_for_assignment_and_notify(self, applet_id: uuid.UUID, respondent_subject_id: uuid.UUID) -> None:
+        assignments = await ActivityAssigmentCRUD(self.session).get_by_respondent_subject_id(respondent_subject_id)
+        if not assignments:
+            return
+
+        activity_ids = []
+        flow_ids = []
+        for assignment in assignments:
+            if assignment.activity_id:
+                activity_ids.append(assignment.activity_id)
+            if assignment.activity_flow_id:
+                flow_ids.append(assignment.activity_flow_id)
+
+        activities = {
+            activity.id: activity
+            for activity in await ActivitiesCRUD(self.session).get_by_applet_id_and_activities_ids(
+                applet_id, activity_ids
+            )
+        }
+        flows = {
+            flow.id: flow for flow in await FlowsCRUD(self.session).get_by_applet_id_and_flows_ids(applet_id, flow_ids)
+        }
+
+        respondent_subject = await SubjectsCrud(self.session).get_by_id(respondent_subject_id)
+        assert respondent_subject
+
+        respondent_activities: dict[uuid.UUID, set[str]] = defaultdict(set)
+        for assignment in assignments:
+            activity_or_flow = activities.get(assignment.activity_id) or flows.get(assignment.activity_flow_id)
+            if activity_or_flow:
+                respondent_activities[respondent_subject_id].add(activity_or_flow.name)
+
+        if len(respondent_activities[respondent_subject_id]) > 0:
+            await self.send_email_notification(
+                applet_id, {respondent_subject_id: respondent_subject}, respondent_activities
+            )
+
+    async def send_email_notification(
+        self,
+        applet_id: uuid.UUID,
+        subjects: dict[uuid.UUID, SubjectSchema],
+        respondent_activities: dict[uuid.UUID, set[str]],
+    ) -> None:
+        service = MailingService()
+        applet = await AppletsCRUD(self.session).get_by_id(applet_id)
+        for respondent_subject_id, activities in respondent_activities.items():
+            respondent_subject: SubjectSchema = subjects[respondent_subject_id]
+
+            language = respondent_subject.language or "en"
+            template_name = self._get_email_template_name(language)
+
+            domain = settings.service.urls.frontend.web_base
+            path = settings.service.urls.frontend.applet_home
+            link = (
+                f"https://{domain}/{path}/{applet.id}"
+                if not domain.startswith("http")
+                else f"{domain}/{path}/{applet.id}"
+            )
```

Comment:
Does this check for `http` in the URL belong here? I believe `settings.service.urls.frontend.web_base` is generally being used throughout the code with the assumption that it does _not_ include any URL schema (`http://`/`https://`). I question whether it makes sense to perform this check only here, if everywhere else, this check is not happening?

I'd say that if we'd like to fix the BE so that it's acceptable for `settings.service.urls.frontend.web_base` to include the URL schema, then I think it will have to be widespread change, removing the `https://` prefix everywhere in the code that `settings.service.urls.frontend.web_base` is being used. Doing it here and _only_ here doesn't make much sense to me.

I'd say don't worry about this for now, and if we want to support `http`, let's do it universally, in a separate ticket where that amount of refactoring is included in the estimate.

```suggestion
            link = f"https://{domain}/{path}/{applet.id}"
```

Reply:
I add this condition only to help testing locally with localhost and will not impact in production and other envs, we have other places that maybe we should change, but in another ticket